### PR TITLE
refactor(textfield): invalid and valid padding inline end 

### PR DESCRIPTION
**Pull request by jenndiaz**
## Description

Address issue with Text field and Text area where input text was appearing behind the icon when the user entered text (see screenshots). 
- added padding inline end to inputs of valid and invalid variants to match designs and address reported issue
- looked into and removed old code as indicated with TODO comments 

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Testing Outline 
1. Open the [docs site](https://pr-2220--spectrum-css.netlify.app/textfield#valid) for text-field. 
- [x] Enter text into alert and valid text fields until input is full, padding should prevent text from going behind the valid/invalid icons 
2. Open the [docs site](https://pr-2220--spectrum-css.netlify.app/textarea#valid) for text-area. 
- [x] Enter text into alert and valid text fields until input is full, padding should prevent text from going behind the valid/invalid icons 

### Regression testing

Validate:

1. A legacy documentation page (such as [accordion](https://pr-###--spectrum-css.netlify.app/accordion.html)), including:

- [ ] The page renders correctly
- [ ] The page is accessible
- [ ] The page is responsive

3. A migrated documentation page (such as [action group](https://pr-###--spectrum-css.netlify.app/actiongroup.html)), including:

- [ ] The page renders correctly
- [ ] The page is accessible
- [ ] The page is responsive

### Before
Screenshot 2023-10-19 at 4 49 58 PM
Screenshot 2023-10-19 at 4 50 18 PM

### After
Screenshot 2023-10-19 at 4 51 15 PM
Screenshot 2023-10-19 at 4 51 46 PM

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.

- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.

- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

**Review by mdt2 — APPROVED** [review]
Looks good to me! Thanks for addressing the TODOs as well.

File: components/textfield/metadata/mods.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 | Modifiable Custom Properties                            |
 | ------------------------------------------------------- |
+| `--mod--textfield-icon-spacing-inline-start-invalid`    |
```
Comment (jawinn) [inline components/textfield/metadata/mods.md:3]: It looks like this mod name in the CSS just needs a dash removed (and the mods.md regenerated).

**Review by jawinn — CHANGES_REQUESTED** [review]
Looks good to me! I tested invalid, valid, large size, and the quiet variants, and the issue has been resolved.

File: components/textfield/metadata/mods.md (hunk 1 above)
Comment (jenndiaz) [reply components/textfield/metadata/mods.md]: good catch! Fixed!

**Review by jawinn — APPROVED** [review]

**Pull request merged**